saleor/graphql/attribute/utils.py

```diff
@@ -370,20 +371,36 @@ def save(cls, instance: T_INSTANCE, cleaned_input: T_INPUT_MAP):
         :param instance: the product or variant to associate the attribute against.
         :param cleaned_input: the cleaned user input (refer to clean_attributes)
         """
+        errors = []
+
         for attribute, attr_values in cleaned_input:
-            if attribute.input_type == AttributeInputType.FILE:
-                attribute_values = cls._pre_save_file_value(
-                    instance, attribute, attr_values
+            try:
+                if attribute.input_type == AttributeInputType.FILE:
+                    attribute_values = cls._pre_save_file_value(
+                        instance, attribute, attr_values
+                    )
+                elif attribute.input_type == AttributeInputType.REFERENCE:
+                    attribute_values = cls._pre_save_reference_values(
+                        instance, attribute, attr_values
+                    )
+                else:
+                    attribute_values = cls._pre_save_values(attribute, attr_values)
+
+                associate_attribute_values_to_instance(
+                    instance, attribute, *attribute_values
                 )
-            elif attribute.input_type == AttributeInputType.REFERENCE:
-                attribute_values = cls._pre_save_reference_values(
-                    instance, attribute, attr_values
+
+            except DataError as e:
+                errors.append(
+                    ValidationError(
+                        str(e),
+                        code=ProductErrorCode.INVALID.value,
+                        params={"attributes": [attr_values.global_id]},
+                    )
                 )
```

Comment:
I don't see the reason to create the same error for every attribute value. I suggest using the list for collecting failing attribute values and then raising one error with the attribute list as params.

Reply:
Sorry, I didn't notice that the message is the error message from the database, so it must be left like that.